x-pack/plugins/security_solution/server/lib/detection_engine/routes/utils.ts

```diff
@@ -158,24 +158,27 @@ const statusToErrorMessage = (statusCode: number) => {
 export class SiemResponseFactory {
   constructor(private response: KibanaResponseFactory) {}
 
-  error<T>({ statusCode, body, headers }: CustomHttpResponseOptions<T>) {
-    const contentType: CustomHttpResponseOptions<T>['headers'] = {
-      'content-type': 'application/json',
-    };
-    const defaultedHeaders: CustomHttpResponseOptions<T>['headers'] = {
-      ...contentType,
-      ...(headers ?? {}),
-    };
-
-    return this.response.custom({
-      headers: defaultedHeaders,
-      statusCode,
-      body: Buffer.from(
+ error<T>({ statusCode,body,headers }: CustomHttpResponseOptions<T>) {
+  const contentType: CustomHttpResponseOptions<T>['headers'] = {
+    'content-type': 'application/json',
+  };
+  const defaultedHeaders: CustomHttpResponseOptions<T>['headers'] = {
+    ...contentType,
+    ...(headers ?? {}),
+  };
+  return this.response.custom({
+    message:statusToErrorMessage(statusCode),
+    error_code:statusCode,
+    attributes:Buffer.from(
         JSON.stringify({
-          message: body ?? statusToErrorMessage(statusCode),
-          status_code: statusCode,
+         status_code:statusCode,
+          message:body,
+          
+          
         })
       ),
+      headers:defaultedHeaders,
+      statusCode:0
```

Comment:
Http responses can only contain a status code, headers and a body, so the attributes will have to be added to the attributes. I don't know why we have `Buffer.from` and `JSON.stringify` it should be possible to just return the body as an object and Core will serialize it.

```
return this.response.custom({
      headers: defaultedHeaders,
      statusCode,
      body: {
        message: body ?? statusToErrorMessage(statusCode),
        status_code: statusCode,
        attributes // you still need to change the error function signature to accept an attributes param
      },
    });
```